src/service_discovery_setup.js

```diff
@@ -0,0 +1,274 @@
+/*
+ * Things Gateway App.
+ *
+ * Back end main script.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+// External dependencies
+const dnssd = require('dnssd');
+const config = require('config');
+const Settings = require('./models/settings');
+
+// Internal dependancies
+
+const server = new DNSserviceDiscovery();
+
+
+/**
+ * Our DNS Service Discovery object.
+ *
+ * Creates an object which contains the 'profile' used to have a mDNS service
+ * discovery process running. The object will be instanciated with default
+ * values to ensure it will always work no matter DB or Config state.
+ *
+ * It provides methods to interact with the object. This object is created
+ * quite early on since the starting process in app.js and parts of the UI in
+ * settings_controller.js need to use its methods.
+ *
+ *
+ * @param {}
+ */
+
+function DNSserviceDiscovery() {
+  this.serviceState = false;
+  this.port = config.get('ports.https');
+  this.localDomain = config.get(
+    'settings.defaults.domain.localControl.mdnsServiceDomain');
+  this.localName= config.get(
```

Comment:
`eslint` issues: you need to add spaces before `=`

Reply:
done for all of them.